src/vs/platform/terminal/node/windowsShellHelper.ts

```diff
@@ -55,34 +51,12 @@ export class WindowsShellHelper extends Disposable implements IWindowsShellHelpe
 		if (this._isDisposed) {
 			return;
 		}
-
-		// The debounce is necessary to prevent multiple processes from spawning when
-		// the enter key or output is spammed
-		Event.debounce(this._onCheckShell.event, (l, e) => e, 150, true)(async () => {
-			await timeout(50);
-			this.checkShell();
-		});
-
-		// We want to fire a new check for the shell on a linefeed, but only
-		// when parsing has finished which is indicated by the cursormove event.
-		// If this is done on every linefeed, parsing ends up taking
-		// significantly longer due to resetting timers. Note that this is
-		// private API.
-		this._xterm.onLineFeed(() => this._newLineFeed = true);
-		this._xterm.onCursorMove(() => {
-			if (this._newLineFeed) {
-				this._onCheckShell.fire(undefined);
-				this._newLineFeed = false;
-			}
-		});
-
-		// Fire a new check for the shell when any key is pressed.
-		this._xterm.onKey(() => this._onCheckShell.fire(undefined));
+		this.checkShell();
 	}
 
-	private checkShell(): void {
+	@throttle(1000)
+	checkShell(): void {
```

Comment:
I think this will run `checkShell` every second if it's called constantly, I wonder if we can have it call 500ms after the first call in a batch and then 500ms after the last call in a batch instead? Say we have a command that runs for 10 seconds, it would be best to only check twice in that time, rather than 10 times.

Reply:
not sure how to determine if it's the first / last call in a batch

Reply:
We may need to do this ourselves instead of using the decorators as it's kind of specialized case, using `setTimeout`/`clearTimeout`